python/mxnet/optimizer.py

```diff
@@ -1076,6 +1076,13 @@ class AdaGrad(Optimizer):
     Methods for Online Learning and Stochastic Optimization*, and available at
     http://www.jmlr.org/papers/volume12/duchi11a/duchi11a.pdf.
 
+    This optimizer updates each weight by::
+    
+            grad = clip(grad * rescale_grad, clip_gradient)
+            history += square(grad)
+            div = grad / sqrt(history + float_stable_eps)
+            weight += (div + weight * wd) * -lr
```

Comment:
This step is problematic. For Adagrad, the stpesize is generally much smaller than the one for SGD, but here you separate the gradient of l2 norm from the div. This can make the training slow if we use a small stepwise, vice versa, a large stepwise can lead to divergence